api/api-config-repos-v2/src/main/java/com/thoughtworks/go/apiv2/configrepos/ConfigReposControllerV2.java

```diff
@@ -53,7 +53,7 @@
 
     @Autowired
     public ConfigReposControllerV2(ApiAuthenticationHelper authHelper, ConfigRepoService service, EntityHashingService entityHashingService) {
-        super(ApiVersion.v2);
+        super(ApiVersion.v2, EntityType.ConfigRepo.name(), "20.2.0", "20.5.0", ApiVersion.v3);
```

Comment:
WDYT about annotating the class as deprecated over passing these deprecation arguments in a constructor.